internal/encryption/method/aesgcm/config.go

```diff
@@ -8,18 +8,124 @@ package aesgcm
 import (
 	"fmt"
 
+	"github.com/opentofu/opentofu/internal/collections"
+
 	"github.com/opentofu/opentofu/internal/encryption/method"
 )
 
+// The following settings follow the NIST SP 800-38D recommendation.
+const (
+	defaultTagSize   = 16
+	defaultNonceSize = 12
+	minimumNonceSize = 1
+)
+
+// validKeyLengths holds the valid key lengths supported by this method.
+var validKeyLengths = collections.Set[int]{
+	16: {},
+	24: {},
+	32: {},
+}
+
+// validTagSizes contains the valid tag sizes according to NIST SP 800-38D 5.2.1.2
```

Comment:
I would not make this configurable as there is no reason to use anything other than the default. 

12 byte tags give a 2^48 probability of collision, and no benefit in performance or storage relative to a state file. The entire tag is still computed and the remainder is truncated before comparison. This is mostly beneficial to resource constrained network protocols, which this is not. 

My 2c.